File: Words_Notes/app/src/main/java/com/example/user/words_notes/MainActivity.java
package com.example.user.words_notes;

import android.content.DialogInterface;
import android.content.Intent;
import android.database.Cursor;
import android.database.sqlite.SQLiteDatabase;
import android.os.Bundle;
import android.support.v7.app.AlertDialog;
import android.support.v7.app.AppCompatActivity;
import android.text.Editable;
import android.text.TextWatcher;
import android.util.Log;
import android.view.Menu;
import android.view.MenuItem;
import android.view.View;
import android.widget.AdapterView;
import android.widget.Button;
import android.widget.EditText;
import android.widget.ListView;
import android.widget.SearchView;
import android.widget.TextView;
import android.widget.Toast;

import com.example.user.LayoutClass.Note;
import com.example.user.LayoutClass.NoteAdapter;
import com.example.user.dao.NotesCRUD;
import com.example.user.dao.NotesDBHelper;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

public class MainActivity extends AppCompatActivity implements View.OnClickListener, AdapterView.OnItemLongClickListener{
    NotesDBHelper notesdb;
    private SQLiteDatabase dbutil;
    private List<Note> noteList = null;
    private String searchStr = "";
    private NotesCRUD noteutill = null;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        //Link database
        notesdb = new NotesDBHelper(this);
        noteutill = new NotesCRUD(notesdb);
        dbutil = notesdb.getReadableDatabase();

        //find all button or ui
        ListView list = (ListView)findViewById(R.id.listView);
        Button newNotes = (Button)findViewById(R.id.new_note);
        final Button folder = (Button)findViewById(R.id.folder);
        TextView folder_name = (TextView)findViewById(R.id.folder_name);

        /*
        Init ListView
         */
        noteList = noteutill.getAll();
        NoteAdapter adapter = new NoteAdapter(MainActivity.this, R.layout.note_item, noteList);

        list.setAdapter(adapter);

        /*
        UPDATE: Click item to edit
         */
        list.setOnItemClickListener(new AdapterView.OnItemClickListener() {
            @Override
            public void onItemClick(AdapterView<?> adapterView, View view, int i, long l) {
                Note note = noteList.get(i);
                TextView folder_name = (TextView)findViewById(R.id.folder_name);
                EditActivity.actionStart(MainActivity.this, folder_name.getText().toString(),
                        note.getTitle(), note.getContent());
                note = noteList.get(i);
                ReflushItemList(noteList);
            }
        });
        list.setOnItemLongClickListener(this);

        //Search content listener
        SearchView searchView= (SearchView) findViewById(R.id.searchView);
        searchView.setOnQueryTextListener(new SearchView.OnQueryTextListener() {
            @Override
            public boolean onQueryTextSubmit(String s) {  // 点击软件盘搜索按钮会弹出 吐司
                Toast.makeText(MainActivity.this, "提交", Toast.LENGTH_SHORT).show();
                return false;
            }
            // 搜索框文本改变事件
            @Override
            public boolean onQueryTextChange(String s) {
                // 文本内容是空就让 recyclerView 填充全部数据 // 可以是其他容器 如listView
                if (s.isEmpty()) {  // 文本工具 检测是否为空，检测空，是输入文本改变 并且为空时触发，刚点击时候虽然为空，但是文本内容没有改变
                    // 设置 容器 的更新
                    searchStr = "";
                    noteList = noteutill.getAll();
                    ReflushItemList(noteList);
                }
                else{
                    searchStr = s;
                    noteList = noteutill.queryByContent(searchStr);
                    ReflushItemList(noteList);
                }
                return false;
            }
        });

        newNotes.setOnClickListener(this);
        folder.setOnClickListener(this);
    }

    public void ReflushItemList(List<Note> noteList){
        ListView list = (ListView)findViewById(R.id.listView);
        NoteAdapter adapter = new NoteAdapter(MainActivity.this, R.layout.note_item, noteList);
        list.setAdapter(adapter);
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        // Inflate the menu; this adds items to the action bar if it is present.
        getMenuInflater().inflate(R.menu.menu_main, menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        // Handle action bar item clicks here. The action bar will
        // automatically handle clicks on the Home/Up button, so long
        // as you specify a parent activity in AndroidManifest.xml.
        int id = item.getItemId();

        //noinspection SimplifiableIfStatement
        if (id == R.id.action_settings) {
            return true;
        }

        return super.onOptionsItemSelected(item);
    }

    public List<Note> getNotesList(){//get notes list from sqlite database -- Notesdb
        List<Note> noteList = new ArrayList<>();
        String sql = "select * from Notes";
        Cursor cursor = dbutil.rawQuery(sql, null);
        Log.d("SQL", sql);
        while(cursor.moveToNext()){
            String title = cursor.getString(cursor.getColumnIndex("title"));
            String content = cursor.getString(cursor.getColumnIndex("context"));
            String date = cursor.getString(cursor.getColumnIndex("createDate"));
            noteList.add(new Note(title, content, date));
        }
        cursor.close();
        return noteList;
    }

    /*
    TODO:
    1. click newNotes to jump to EditActivity. Finished!
    2. click item once to get the context and edit it (update). Finished!
    3. click item long time to remove the item or edit it (delete or update)
    4. input string to search notes, using like and % %. Finished!
    5. click folder to jump to ShowFolderActivity.
     */
    @Override
    public void onClick(View view) {
        switch (view.getId()){
            case R.id.new_note:{
                TextView folder_name = (TextView)findViewById(R.id.folder_name);
                EditActivity.actionStart(MainActivity.this, folder_name.getText().toString());
            }break;
            case R.id.folder:{

            }break;
            default:
                break;
        }
    }

    @Override
    public boolean onItemLongClick(AdapterView<?> adapterView, View view, int i, long l) {
        final int ITEM = i;
        final long L = l;
        TextView text = (TextView)findViewById(R.id.folder_name);
        final String folder = text.getText().toString();
        final String title = noteList.get(ITEM).getTitle();
        AlertDialog.Builder builder = new AlertDialog.Builder(this);
        builder.setTitle("删除");
        builder.setMessage("该操作不可逆,确认删除吗？");
        builder.setPositiveButton("确定", new DialogInterface.OnClickListener() {
            @Override
            public void onClick(DialogInterface dialogInterface, int i) {
                noteutill.delete(folder, title);
                noteList = getNotesList();
                ReflushItemList(noteList);
            }
        });
        builder.setNegativeButton("取消", new DialogInterface.OnClickListener() {
            @Override
            public void onClick(DialogInterface dialogInterface, int i) {
            }
        });
        builder.create();
        builder.show();
        return false;
    }
}

File: calculate/app/src/main/java/com/example/user/calculate/MainActivity.kt
package com.example.daisy.calculate


import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.TextView
import android.widget.Toast

class MainActivity : AppCompatActivity(), View.OnClickListener {
    internal var input: TextView? = null  //初始化
    internal var clear_flag: Boolean = false //初始化

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        input = findViewById(R.id.input) as TextView
        //数字实例化
        val button_0 = findViewById(R.id.button_0) as Button
        val button_1 = findViewById(R.id.button_1) as Button
        val button_2 = findViewById(R.id.button_2) as Button
        val button_3 = findViewById(R.id.button_3) as Button
        val button_4 = findViewById(R.id.button_4) as Button
        val button_5 = findViewById(R.id.button_5) as Button
        val button_6 = findViewById(R.id.button_6) as Button
        val button_7 = findViewById(R.id.button_7) as Button
        val button_8 = findViewById(R.id.button_8) as Button
        val button_9 = findViewById(R.id.button_9) as Button

        //操作实例化
        val point = findViewById(R.id.point) as Button
        val clear = findViewById(R.id.clear) as Button
        val delet = findViewById(R.id.delet) as Button
        val percent = findViewById(R.id.percent) as Button
        val divide = findViewById(R.id.divide) as Button
        val mul = findViewById(R.id.mul) as Button
        val sub = findViewById(R.id.sub) as Button
        val add = findViewById(R.id.add) as Button
        val equ = findViewById(R.id.equ) as Button


        button_0.setOnClickListener(this)
        button_1.setOnClickListener(this)
        button_2.setOnClickListener(this)
        button_3.setOnClickListener(this)
        button_4.setOnClickListener(this)
        button_5.setOnClickListener(this)
        button_6.setOnClickListener(this)
        button_7.setOnClickListener(this)
        button_8.setOnClickListener(this)
        button_9.setOnClickListener(this)
        point.setOnClickListener(this)
        clear.setOnClickListener(this)
        delet.setOnClickListener(this)
        add.setOnClickListener(this)
        point.setOnClickListener(this)
        mul.setOnClickListener(this)
        divide.setOnClickListener(this)
        input!!.setOnClickListener(this)
        percent.setOnClickListener(this)
    }

    override fun onClick(v: View) {
        var str = input!!.text.toString()
        when (v.id) {
            R.id.button_0, R.id.button_1, R.id.button_2, R.id.button_3, R.id.button_4, R.id.button_5, R.id.button_6, R.id.button_7, R.id.button_8, R.id.button_9, R.id.point -> {
                if (clear_flag) {
                    clear_flag = false
                    str = ""
                }
                input!!.text = str + (v as Button).text
            }
            R.id.sub, R.id.divide, R.id.mul, R.id.add -> {
                clear_flag = false
                input!!.text = str + " " + (v as Button).text + " "//在屏幕显示+，-，*，/，并于字符间有空格
            }
            R.id.percent -> {
                val result3 = 0.01 * java.lang.Double.parseDouble(str)
                val r3 = result3.toString()
                input!!.text = r3
            }

            R.id.delet -> input!!.text = str.substring(0, str.length - 1)//从字符串的0位置开始，索引到字符串的倒数第二个字符串
            R.id.clear -> {
                clear_flag = false
                input!!.text = ""
            }
            R.id.equ -> getResult()
        }
    }

    private fun getResult() {
        val exp = input!!.text.toString()

        if (!exp.contains(" ")) {
            return
        }
        clear_flag = true
        var result = 0.0
        val s1 = exp.substring(0, exp.indexOf(" ")) //获取运算符前面的字符串
        val op = exp.substring(exp.indexOf(" ") + 1, exp.indexOf(" ") + 2)
        val s2 = exp.substring(exp.indexOf(" ") + 3)//获取运算符后面的字符串
        if (s1 != " " && s2 != " ") {
            var d1 = java.lang.Double.parseDouble(s1)
            var d2 = java.lang.Double.parseDouble(s2)
            when (op) {
                "+" -> result = d1 + d2
                "-" -> result = d1 - d2
                "*" -> result = d1 * d2
                "/" -> if (d2 == 0.0) {
                    Toast.makeText(this@MainActivity, "除数不能为0", Toast.LENGTH_SHORT).show()
                } else {
                    result = d1 / d2
                }
            }
            //判断第一个字符是否为小数点，如果是小数点自动变为0.x,如果用正负号转换，自动变为-0.x
            if (s1[0] == '.') {
                d1 = 0 + d1
            }
            if (s2[0] == '.') {
                d2 = 0 + d2
            }
            if (s1[0] != '-') {
                d1 = 0 - d1
            }

            input!!.text = result.toString()//把运算结果输出到显示屏

        } else if (s1[0] != '-') {
            val s = exp.substring(2, exp.indexOf(" "))
            val d1 = 0 - java.lang.Double.parseDouble(s)
        } else if (s2[0] != '-') {
            val s = exp.substring(exp.indexOf(" ") + 4)
            val d2 = 0 - java.lang.Double.parseDouble(s)
        } else
            return

    }
}
